### ISSUE TITLE
JOSS Review: General documentation suggestions.

### ISSUE THREAD

**Issue by JonasMoss** [KaiWenger/memochange#7]
See also https://github.com/KaiWenger/memochange/issues/3.

Here are some suggestions for the documentation, most of them are not critical.

* Better documented return values: 
  * "Returns the test statistic and the corresponding critical values of the test." Should read as "numeric vector" containing the 90% 95% 99% and the test statistic.  
  * Have you considered returning objects modeled after the functions of the `stats` package? For instance, the `stats::t_test` function returns an object instead of a vector. This is probably a better choice since it allows you to add information to the vector in the future without breaking the API.
* Better documented `x` as numeric time series; what does this mean? Is it a 
  vector, a `zoo` object, or something else? For instance `LongMemoryTS::local.W` says "vector of length T", which is more understandable. This point is important since it tells me what kind of input the program should work on. 
* Document that tau is in [0, 1] in all relevant functions and throw an error if it is outside the bounds. When `tau` is in [0,0.5], throw an error if the user provides a `tau` outside these bounds. I'm adding this as a separate point since it is unexpected for the user that the domain of definition for `tau` changes across functions.
* **Examples**:
  * Please provide short comments about what you are doing in the examples. 
  * For reproducibility use `set.seed` when simulating data in the examples. 
  * There should preferably be more substantial examples using real data sets. There are two possibilities here:
    1. Use data sets from other packages, e.g. `datasets` or one your dependencies.
    2. Add your own data. 
* Add `seealso{}` to your documentation, e.g. from `snsupwald`to `
snwilcoxon`.

**Nitpicking**:
  * Be consistent in name of the time series: Sometimes called `x`, othertimes `tseries`.
  * tau is defined slightly different in `BP_estim` vs eg. CUSUMfixed, is this intended? In `BP_estim` it references `T`, but not in `CUSUMfixed`.
  * Lint the example code for stylistic errors such as `series<- c(rnorm(200),cumsum(rnorm(200)))` which should be `series <- c(rnorm(200), cumsum(rnorm(200)))`.

**Comment by KaiWenger** [comment]
Hi @JonasMoss,
we incorporated your remarks as follows:

- we documented our return values in a better way. We heavily thought about returning objects modeled after the functions of the `stats` package, but decided to skip this.
- we documented `x` as numeric time series and test for it to be a numeric time series in all of our functions.
- we changed the documentation about `tau`. We included an `error` message in all functions if `tau` is chosen in an incorrect manner. We further included a  `warning` message if the chosen `tau` is correct, but does not match the critical values deposited in the package.
- we (a) included short comments to our examples, (b) use the `set.seed` for reproducibility and (c) add `seealso{}` to our documentation.
- in the examples of the implemented functions we stayed with the simulated time series since then the user exactly knows how the tests should behave. However, we included an additional example in one of the two new vignettes and in the `README`. We further explained the real data examples in more detail there.
- we changed the arguments of part of the functions such that the name of the time series is always `x`.
- we named both `tau` since it is usual in literature. We hope that we changed the explanation in the arguments in a way such that no misunderstanding happens.
- we hope that we fixed all stylistic errors in the examples of all functions.

Once again, thank you very much for all these helpful and constructive suggestions!
